internal/elasticsearch/ingest/datastream.go

```diff
@@ -16,12 +16,25 @@ import (
 	"strings"
 	"time"
 
+	"gopkg.in/yaml.v3"
+
 	"github.com/elastic/elastic-package/internal/elasticsearch"
 	"github.com/elastic/elastic-package/internal/packages"
 )
 
 var ingestPipelineTag = regexp.MustCompile(`{{\s*IngestPipeline.+}}`)
 
+type Rule struct {
+	TargetDataset interface{}   `yaml:"target_dataset"`
+	If            interface{}   `yaml:"if"`
+	Namespace     []interface{} `yaml:"namespace"`
+}
+
+type SourceData struct {
+	SourceDataset interface{} `yaml:"source_dataset"`
```

Comment:
According to the the package-spec 2.9.0, the [source_dataset](https://github.com/elastic/package-spec/blob/067a0e5656a1603d03ff7f7589c1d9517becca3d/spec/integration/data_stream/routing_rules.spec.yml#L64-L68) is a `string`. Do you think we use the `string` type here, or there's a reason to use `inerface{}`?